loki/frontend/fparser.py

```diff
@@ -229,6 +229,41 @@ def rget_child(node, node_type):
     return None
 
 
+def _get_comments_from_section(sec, include_pragmas=False, reverse=False):
```

Comment:
You could consider making this a common frontend utility. We're doing the same hack for OMNI: https://github.com/ecmwf-ifs/loki/blob/main/loki/frontend/omni.py#L495